sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqMessage.java

```diff
@@ -0,0 +1,297 @@
+      Integer deliveryMode,
+      Integer priority,
+      String correlationId,
+      String replyTo,
+      String expiration,
+      String messageId,
+      Date timestamp,
+      String type,
+      String userId,
+      String appId,
+      String clusterId) {
+    this.routingKey = routingKey;
+    this.body = body;
+    this.contentType = contentType;
+    this.contentEncoding = contentEncoding;
+    this.headers = headers;
+    this.deliveryMode = deliveryMode;
+    this.priority = priority;
+    this.correlationId = correlationId;
+    this.replyTo = replyTo;
+    this.expiration = expiration;
+    this.messageId = messageId;
+    this.timestamp = timestamp;
+    this.type = type;
+    this.userId = userId;
+    this.appId = appId;
+    this.clusterId = clusterId;
+  }
+
+  public String getRoutingKey() {
+    return routingKey;
+  }
+
+  public byte[] getBody() {
+    return body;
+  }
+
+  public String getContentType() {
+    return contentType;
+  }
+
+  public String getContentEncoding() {
+    return contentEncoding;
+  }
+
+  public Map<String, Object> getHeaders() {
+    return headers;
+  }
+
+  public Integer getDeliveryMode() {
+    return deliveryMode;
+  }
+
+  public Integer getPriority() {
+    return priority;
+  }
+
+  public String getCorrelationId() {
+    return correlationId;
+  }
+
+  public String getReplyTo() {
+    return replyTo;
+  }
+
+  public String getExpiration() {
+    return expiration;
+  }
+
+  public String getMessageId() {
+    return messageId;
+  }
+
+  public Date getTimestamp() {
+    return timestamp;
+  }
+
+  public String getType() {
+    return type;
+  }
+
+  public String getUserId() {
+    return userId;
+  }
+
+  public String getAppId() {
+    return appId;
+  }
+
+  public String getClusterId() {
+    return clusterId;
+  }
+
+  public AMQP.BasicProperties createProperties() {
+    return new AMQP.BasicProperties()
+        .builder()
+        .contentType(contentType)
+        .contentEncoding(contentEncoding)
+        .headers(headers)
+        .deliveryMode(deliveryMode)
+        .priority(priority)
+        .correlationId(correlationId)
+        .replyTo(replyTo)
+        .expiration(expiration)
+        .messageId(messageId)
+        .timestamp(timestamp)
+        .type(type)
+        .userId(userId)
+        .appId(appId)
+        .clusterId(clusterId)
+        .build();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        routingKey,
+        Arrays.hashCode(body),
+        contentType,
+        contentEncoding,
+        deliveryMode,
+        priority,
+        correlationId,
+        replyTo,
+        expiration,
+        messageId,
+        timestamp,
+        type,
+        userId,
+        appId,
+        clusterId);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj instanceof RabbitMqMessage) {
+      RabbitMqMessage other = (RabbitMqMessage) obj;
+      if (routingKey == null
+          && other.routingKey == null
```

Comment:
Can't this be handled by the other branch equally well? I don't think that having these fields be null is a special case - you already handle nulls below.